### PR TITLE
Refactor the mempool error docs

### PR DESCRIPTION
**Pull request by upbqdn**
This is a small PR that attempts to resolve this conversation https://github.com/ZcashFoundation/zebra/pull/2942/files#r735892194.

## Review

@teor2345 has probably most of the context here, but anyone else can review.

### CODE REVIEW

File: zebrad/src/components/mempool/error.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ pub enum MempoolError {
     ///
     /// Note that the mempool caches this error. See [`super::storage::Storage`]
     /// for more details.
-    #[error("mempool storage has a cached tip rejection for this exact transaction")]
+    #[error("the authorizing data of the transaction conflicts with the current chain tip")]
```
Comment (teor2345) [inline zebrad/src/components/mempool/error.rs]: I want to say something like "this exact transaction" or "identical transactions" here.
What do you think?

```suggestion
    #[error("the transaction will be rejected from the mempool until the next chain tip block")]
```

File: zebrad/src/components/mempool/error.rs [hunk 2]
```diff
@@ -33,9 +33,7 @@ pub enum MempoolError {
     ///
     /// Note that the mempool caches this error. See [`super::storage::Storage`]
     /// for more details.
-    #[error(
-        "mempool storage has a cached tip rejection for any transaction with the same effects"
-    )]
+    #[error("the effects of the transaction conflict with the current chain tip")]
```
Comment (teor2345) [inline zebrad/src/components/mempool/error.rs]:
```suggestion
    #[error("any transaction with the same effects will be rejected from the mempool until the next chain tip block")]
```

File: zebrad/src/components/mempool/error.rs [hunk 3]
```diff
@@ -46,9 +44,7 @@ pub enum MempoolError {
     ///
     /// Note that the mempool caches this error. See [`super::storage::Storage`]
     /// for more details.
-    #[error(
-        "mempool storage has a cached chain rejection for any transaction with the same effects"
-    )]
+    #[error("the effects of the transaction conflict with the current chain")]
```
Comment (teor2345) [inline zebrad/src/components/mempool/error.rs]:
```suggestion
    #[error("any transaction with the same effects will be rejected from the mempool until a chain reset")]
```

**Review by teor2345 — COMMENTED** [review]
This is a good start.

Sometimes we reject transactions because they are invalid, rather than due to a conflict. (For example, an invalid signature.)

I made some suggestions with more generic language.

**Comment by upbqdn** [comment]
Those were great suggestions. Thank you. They all make sense to me now.

**Review by conradoplg — APPROVED** [review]
It seems much clearer :+1:

**Pull request merged**